Cloth.Application/ClothService.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Cloth.Application.Models;
+using Cloth.Application.Models.Responses;
+using Microsoft.Extensions.Logging;
+
+namespace Cloth.Application;
+
+using Cloth.Application.Extensions;
+using Cloth.Domain.Entities;
+using Cloth.Domain.Exceptions;
+
+public class ClothService : IClothService
+{
+    private readonly IClothRepository clothRepository;
+    private readonly ILogger<ClothService> _logger;
+    public ClothService(IClothRepository memberRepository, ILogger<ClothService> logger)
+    {
+        _logger = logger;
+        this.clothRepository = memberRepository;
+    }
+
+    /// <summary>
+    /// Gets all Cloths from the repository
+    /// </summary>
+    /// <returns></returns>
+    public async Task<IEnumerable<Cloth>> GetAllCloths()
+    {
+        return await clothRepository.GetAllCloths();
+    }
+
+    /// <summary>
+    /// Splits the input to form list of different highlights
+    /// </summary>
+    /// <param name="highlight"></param>
+    /// <returns></returns>
+    public List<string> GetHighlights(string? highlight)
+    {
+        var highlights = new List<string>();
+        if (string.IsNullOrWhiteSpace(highlight))
+        {
+            return highlights;
+        }
+        else if (highlight.Contains(','))
+        {
+            _logger.LogInformation("Multiple highlights detected in " + highlight);
+            highlights = highlight.Split(',').ToList();
+        }
+        else
+        {
+            highlights.Add(highlight);
+        }
+
+        return highlights;
+    }
+
+    public async Task<ResponseDto> FilterClothsAsync(decimal? minPrice, decimal? maxPrice, string? size, string? highlight)
+    {
+        try
+        {
+            var allItems = await clothRepository.GetAllCloths();
+            LoggingExtensions.LogGettingCloths(_logger);
+
+            if (allItems == null || allItems.Count() == 0)
```

Comment:
Because is IEnumerable, you do not have indexes.  This means you use Count() from Linq. This is not good because you can allow this problem to occur "Possible multiple enumeration of IEnumerable".
That's why materialize the collection with .ToLilist();
and then use from the ICollection property .Count to eliminate the problem